auth/src/main/java/com/firebase/ui/auth/data/model/FirebaseUiNetworkException.java

```diff
@@ -0,0 +1,10 @@
+package com.firebase.ui.auth.data.model;
+
+/**
+ * Represents an error connecting to the internet.
+ */
+public class FirebaseUiNetworkException extends FirebaseUiException {
```

Comment:
I think just having a `code` on `FirebaseUiException` is probably better than subclasses.  Checking an `int` or `String` is easier than using `instanceof` and faster too.

I'd generally only use subclasses if extra functionality is needed, like accepting a more advanced parameter.

Wdyt?

Reply:
Oh, I see. Yep, that makes sense but do you agree that `IdpResponse` should eventually deprecate and remove the error codes getter in favor getting the exception (which has that info and more if relevant)?

Reply:
Oh sweet, it turns out `Exception` does some black magic to serialize subclass fields too so we don't lose our error code when passing the exception through an `Intent`. 👍

Reply:
Probably yes but for now we expose so little information in errors that we can stick with the code just to keep the API unchanged.

Reply:
Yes, yes of course. I keep forgetting that my PR was too big for you to take a look at in-depth. There are cool exceptions in that PR such as `MisconfiguredProviderException`, `ExternalProviderError`, etc... we'll be able to provide much more useful information! 😄